core/controllers/reader.py

```diff
@@ -227,12 +228,22 @@ def get(self, exploration_id):
         except Exception as e:
             raise self.PageNotFoundException(e)
 
+        states = {}
+        for state_name in exploration.states:
+            state_dict = exploration.states[state_name].to_dict()
+            state_dict['unresolved_answers'] = (
+                stats_services.get_top_unresolved_answers_for_default_rule(
+                    exploration_id, state_name))
+            states[state_name] = state_dict
         self.values.update({
             'can_edit': (
                 self.user_id and
                 rights_manager.Actor(self.user_id).can_edit(
                     feconf.ACTIVITY_TYPE_EXPLORATION, exploration_id)),
             'exploration': exploration.to_player_dict(),
+            'exploration_id': exploration_id,
+            'init_state_name': exploration.init_state_name,
+            'states': states,
```

Comment:
Yeah, I agree with you this really shouldn't be done (even temporarily to be followed up in an upcoming PR). It's adding extra bloat to the payload for every load of the learner view. We probably don't need any of these three if we update endpoints to access the attached exploration, instead.